tag-expressions/ruby/lib/cucumber/tag_expressions/parser.rb

```diff
@@ -0,0 +1,120 @@
+require 'cucumber/tag_expressions/expressions.rb'
+
+# Array monkey-patch
+class Array
```

Comment:
We load Cucumber into the same Ruby process as other people's code, so we try to avoid adding monkey patches (there is some legacy code that does, but it's something we don't do in new code).

I suggest creating a decorator using [SimpleDelegator](https://hashrocket.com/blog/posts/using-simpledelegator-for-your-decorators) that you can use to create Arrays decorated with this method. You could call it `PoppableArray` or something.